### DISCUSSION TITLE
How to listen for geo-fence events?

### DISCUSSION THREAD

**Discussion by jvalrog** [dgreif/ring#1233] [General]
I want to automatically enable/disable modes based on geofence events.

Is there a way to check for those events?

**Comment by dgreif** [comment]
I have not looked into this API at all, and I'm not sure if they can be received by the client or not. I'd gladly take a look at a PR if someone wants to do the work of reverse engineering these 😄

**Reply by tsightler** [reply]
I've poked around with it a little bit, there is an API call that the app can use to check the geo-fence configuration, and you can get arm/disarm reminders, which seems to be just another type of push notification.  However, the geo-fence settings for these notifications are device specific, i.e. enabling notifications on one device does not enable notifcations on other devices, thus ring-client-api would need to enable these alerts during startup, but I think it could receive them after that point.  Other than that, I'm not sure there is another way to get such events, but this was admittedly just a cursory glance at the feature.

I already have another app I use for geo-fencing, but it might be nice to get rid of it and just use the Ring app, so I might poke around at this a little deeper in the coming weeks, if I can somehow find time.
